### PR TITLE
Fix router tests

### LINKED ISSUE #102: Router test failure
https://travis-ci.com/SkycoinProject/skywire-mainnet/builds/142666285

```
2019-12-30T17:22:08Z] INFO [dmsg_client]: findOrConnectToServers: found/connected to server 021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6948[2019-12-30T17:22:08Z] INFO [dmsg_client]: found dmsg_server entries:[	version: 0.0.1
6949	sequence: 0
6950	registered at: 1577726527553886240
6951	static public key: 021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6952	signature: 6aac45f658501e833978069e8fc501f305892fdd2c56bec6065d9883685179570ab9c972b7147f7344caeb8e917acbe3569adb5f8a18ca0d2c160cdd20e8b71d00
6953	entry is registered as server. Related info: 
6954		address: 127.0.0.1:37787
6955		port: 
6956		available connections: 10
6957
6958]
6959[2019-12-30T17:22:09Z] INFO [noise]: accepted: 02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3
6960[2019-12-30T17:22:09Z] INFO [dmsg_server]: newConn: 127.0.0.1:57540(02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3)
6961[2019-12-30T17:22:09Z] INFO [dmsg_server]: ServingConn connCount=3 srcClient=02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3
6962[2019-12-30T17:22:09Z] INFO [dmsg_client]: findOrConnectToServers: found/connected to server 021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6963[2019-12-30T17:22:09Z] INFO [dmsg_client]: ServingConn connCount=4 remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6964[2019-12-30T17:22:09Z] INFO [tp_manager]: listening on network: dmsg
6965[2019-12-30T17:22:09Z] INFO [tp_manager]: transport manager is serving.
6966[2019-12-30T17:22:09Z] INFO [tp_manager]: listening on network: dmsg
6967[2019-12-30T17:22:09Z] INFO [tp_manager]: transport manager is serving.
6968[2019-12-30T17:22:09Z] INFO [tp_manager]: transport manager is closing.
6969[2019-12-30T17:22:09Z] INFO [tp_manager]: transport manager closed.
6970[2019-12-30T17:22:09Z] INFO [tp_manager]: transport manager is closing.
6971[2019-12-30T17:22:09Z] INFO [tp_manager]: transport manager closed.
6972[2019-12-30T17:22:09Z] INFO [dmsg_client]: ClosingConnection remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6973[2019-12-30T17:22:09Z] INFO [dmsg_client]: ConnectionClosed connCount=3 error="read failed: read tcp 127.0.0.1:57538->127.0.0.1:37787: use of closed network connection" remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6974[2019-12-30T17:22:09Z] DEBUG [dmsg_client]: connection with server closed reason="read failed: read tcp 127.0.0.1:57538->127.0.0.1:37787: use of closed network connection" remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6975[2019-12-30T17:22:09Z] INFO [dmsg_client]: updatingEntry: 	version: 0.0.1
6976	sequence: 0
6977	registered at: 1577726528760718315
6978	static public key: 0345327088f0f34c359ecf67892fbe58a1cd7299e15a797339fa29365cc2e7c551
6979	signature: 5aa08e431abee7f7de3d644343680210f1c17e73bfeeeb80e44f9de341eabf6d515439b8c853b9ccaf9ba378b7b7faeacb2429d861afcf233bcbb74a6804588000
6980	entry is registered as client. Related info: 
6981		delegated servers: 
6982
6983
6984[2019-12-30T17:22:09Z] INFO [dmsg_server]: ClosingConn connCount=2 error="read failed: EOF" srcClient=0345327088f0f34c359ecf67892fbe58a1cd7299e15a797339fa29365cc2e7c551
6985[2019-12-30T17:22:09Z] INFO [dmsg_server]: connection with client 0345327088f0f34c359ecf67892fbe58a1cd7299e15a797339fa29365cc2e7c551 closed: error(read failed: EOF)
6986[2019-12-30T17:22:09Z] INFO [dmsg_client]: ClosingConnection remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6987[2019-12-30T17:22:09Z] INFO [dmsg_client]: ConnectionClosed connCount=1 error="read failed: read tcp 127.0.0.1:57540->127.0.0.1:37787: use of closed network connection" remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6988[2019-12-30T17:22:09Z] DEBUG [dmsg_client]: connection with server closed reason="read failed: read tcp 127.0.0.1:57540->127.0.0.1:37787: use of closed network connection" remoteServer=021584f99b9dc3dfa757a0760c93e0bf44b443dacb474376b90eec7d32a2092865
6989[2019-12-30T17:22:09Z] INFO [dmsg_client]: updatingEntry: 	version: 0.0.1
6990	sequence: 0
6991	registered at: 1577726529086839150
6992	static public key: 02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3
6993	signature: c1e5ae58ec825cf03ae50e8ae8f20794f83da6e13b5b3a0e3fc13f417d854244523e32c12c4d9602fdd94047dafa03dbd7d1e215676038533d88e32a37247ccc01
6994	entry is registered as client. Related info: 
6995		delegated servers: 
6996
6997
6998[2019-12-30T17:22:09Z] INFO [dmsg_server]: ClosingConn connCount=0 error="read failed: EOF" srcClient=02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3
6999[2019-12-30T17:22:09Z] WARN [dmsg_server]: Failed to close connection error="close tcp4 127.0.0.1:37787->127.0.0.1:57540: use of closed network connection" srcClient=02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3
7000[2019-12-30T17:22:09Z] INFO [dmsg_server]: connection with client 02d75d089b307032a3dfd1e6808f6a4ea0011205f60e7f28f69f444c4e48b2b7c3 closed: error(read failed: EOF)
7001[2019-12-30T17:22:09Z] WARN [router-gateway]: Request completed with error. error="error"
7002[2019-12-30T17:22:09Z] WARN [router-gateway]: Request completed with error. error="error"
7003[2019-12-30T17:22:09Z] WARN [router-gateway]: Request completed with error. error="error"
7004
7005FAIL
7006coverage: 75.4% of statements
7007FAIL	github.com/SkycoinProject/skywire-mainnet/pkg/router	64.437s
```

### PR DESCRIPTION
**Pull request by Darkren**
Fixes #102 

 Changes:	
- Uncommented and fixed `TestArbitrarySizeOneMessage`
- Uncommented and fixed `TestArbitrarySizeMultipleMessagesByChunks`
- Fixed `testPresentTimeout` subtest of `TestConn`
- Fixed possible memory leak in route group. Described in details in commit message https://github.com/SkycoinProject/skywire-mainnet/commit/42b955973bb12e7a57ce94ac9b3841cd828fd414
- Fixed `testCloseTimeout` subtest of `TestConn`
- Improved performance of route group (we've been holding mutex much longer than we actually needed it)
- Refactored route group and its tests a bit, cleaned up the code
- Removed debug logs

### CODE REVIEW

**Comment by Darkren** [comment]
Just one more subtest of `TetsConn` left to be fixed. And code needs to be cleaned up

File: pkg/router/route_group_test.go [hunk 1]
```diff
@@ -21,120 +23,51 @@ import (
...
 
-	err = rg0.Close()
+	err := rg1.Close()
 	require.NoError(t, err)
-	require.True(t, rg0.isClosed())
-	require.True(t, rg1.isRemoteClosed())
+	require.True(t, rg1.isClosed())
+	require.True(t, rg2.isRemoteClosed())
 	// rg1 should be done (not getting any new data, returning `io.EOF` on further reads)
 	// but not closed
```
Comment (evanlinjin) [inline pkg/router/route_group_test.go:45]: Should be specify:
```golang
// rg1 should be closed for reads (returning `io.EOF` on further reads), but not on writes.
```
Is this the intended behaviour?

**Review by evanlinjin — APPROVED** [review]
Awesome work! Just a small query for clearer commenting.

File: pkg/router/route_group.go [hunk 2]
```diff
@@ -221,22 +278,32 @@ func (rg *RouteGroup) Write(p []byte) (n int, err error) {
 			return 0, err
 		}
 
+		fmt.Println()
```
Comment (nkryuchkov) [inline pkg/router/route_group.go]: Did you forget to delete it?

**Review by nkryuchkov — APPROVED** [review]
Great job!

File: pkg/router/route_group_test.go (hunk 1 above)
Comment (Darkren) [reply pkg/router/route_group_test.go:45]: @evanlinjin yep. `io.EOF` is only returned when the connection is broken (we test this via `isRemoteClosed`) and no data left on the wire (we check the `readCh` for it). this was the only way to make route group behave like the usual connection, and also it seems like the most correct one to me

File: pkg/router/route_group.go (hunk 2 above)
Comment (Darkren) [reply pkg/router/route_group.go]: Can't even say what was that for lol, but yes, seems so

**Review by Kifen — APPROVED** [review]
Good job!

**Pull request merged**